source/common/network/listen_socket_impl.h

```diff
@@ -33,8 +33,10 @@ class ListenSocketImpl : public ListenSocket {
  */
 class TcpListenSocket : public ListenSocketImpl {
 public:
+  TcpListenSocket(const std::string& address, bool bind_to_port);
   TcpListenSocket(uint32_t port, bool bind_to_port);
```

Comment:
This is for admin config, we are still using port there since it's for local. We can defer the update of admin in a follow up PR if necessary.